File: centralizer/src/main/java/com/example/centralizer/models/compteCourantDTO/TransfertAvecFrais.java
package com.example.centralizer.models.compteCourantDTO;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * Modèle représentant un transfert avec les frais associés
 */
public class TransfertAvecFrais {
    private String idTransfert;
    private String envoyer;
    private String receveur;
    private BigDecimal montant;
    private LocalDate dateTransfert;
    
    // Informations sur les frais
    private BigDecimal fraisEnvoyeur;
    private String libelleFraisEnvoyeur;
    private BigDecimal montantTotalEnvoyeur; // montant + frais
    
    // Constructeur par défaut
    public TransfertAvecFrais() {
    }

    // Getters et setters
    public String getIdTransfert() {
        return idTransfert;
    }

    public void setIdTransfert(String idTransfert) {
        this.idTransfert = idTransfert;
    }

    public String getEnvoyer() {
        return envoyer;
    }

    public void setEnvoyer(String envoyer) {
        this.envoyer = envoyer;
    }

    public String getReceveur() {
        return receveur;
    }

    public void setReceveur(String receveur) {
        this.receveur = receveur;
    }

    public BigDecimal getMontant() {
        return montant;
    }

    public void setMontant(BigDecimal montant) {
        this.montant = montant;
    }

    public LocalDate getDateTransfert() {
        return dateTransfert;
    }

    public void setDateTransfert(LocalDate dateTransfert) {
        this.dateTransfert = dateTransfert;
    }

    public BigDecimal getFraisEnvoyeur() {
        return fraisEnvoyeur;
    }

    public void setFraisEnvoyeur(BigDecimal fraisEnvoyeur) {
        this.fraisEnvoyeur = fraisEnvoyeur;
    }

    public String getLibelleFraisEnvoyeur() {
        return libelleFraisEnvoyeur;
    }

    public void setLibelleFraisEnvoyeur(String libelleFraisEnvoyeur) {
        this.libelleFraisEnvoyeur = libelleFraisEnvoyeur;
    }

    public BigDecimal getMontantTotalEnvoyeur() {
        return montantTotalEnvoyeur;
    }

    public void setMontantTotalEnvoyeur(BigDecimal montantTotalEnvoyeur) {
        this.montantTotalEnvoyeur = montantTotalEnvoyeur;
    }
}

File: ServeurCompteDepot/models/TransfertAvecFrais.cs
using System.ComponentModel.DataAnnotations;
using ServeurCompteDepot.Models;

namespace ServeurCompteDepot.models
{
    /// <summary>
    /// Modèle représentant un transfert avec les frais associés
    /// </summary>
    public class TransfertAvecFrais
    {
        public int IdTransfert { get; set; }
        
        [Required]
        public string CompteEnvoyeur { get; set; } = string.Empty;
        
        [Required]
        public string CompteReceveur { get; set; } = string.Empty;
        
        [Required]
        public decimal Montant { get; set; }
        
        [Required]
        public DateTime DateTransfert { get; set; }
        
        // Informations sur les frais
        public decimal FraisEnvoyeur { get; set; }
        public string LibelleFraisEnvoyeur { get; set; } = string.Empty;
        public decimal MontantTotalEnvoyeur { get; set; } // montant + frais
        
        public TransfertAvecFrais()
        {
        }
        
        // Constructeur avec transfert de base
        public TransfertAvecFrais(Transfert transfert)
        {
            IdTransfert = transfert.IdTransfert;
            CompteEnvoyeur = transfert.Envoyer;
            CompteReceveur = transfert.Receveur;
            Montant = transfert.Montant;
            DateTransfert = transfert.DateTransfert;
        }
    }
}